docs/source/developer/repomap.rst

```diff
@@ -43,8 +43,6 @@ Documentation
 - :ghfile:`README.rst` - README for repo, also uploaded to PyPI
 - :ghfile:`CONTRIBUTING.md` - Documentation on how to contribute to project
   (out of date, should be updated to point to Sphinx docs)
-- :ghfile:`CHANGE_LOG` - History of Numba releases, also directly embedded
-  into Sphinx documentation
```

Comment:
I am still debating whether it is a good idea to remove this file and what can be the alternative.